src/session/session_dhandle.c

```diff
@@ -211,7 +211,13 @@ __wt_session_lock_dhandle(WT_SESSION_IMPL *session, uint32_t flags, bool *is_dea
 
         /* Give other threads a chance to make progress. */
         WT_STAT_CONN_INCR(session, dhandle_lock_blocked);
+
+        /* FIXME-WT-12037 Use a sleep to work around a Windows-specific scheduling issue. */
+#ifdef _WIN32
+        __wt_sleep(0, 1000);
```

Comment:
The last time I tried a patch build with 1 microsecond, the issue didn't reproduce. We can use 1 microsecond instead of 1 millisecond. 

Also, it is possible to use this solution as a permanent fix for all platforms. This code flow gets executed only when multiple threads are trying to open the same dhandle.

Reply:
Sure - but a yield is more explicit in its intent so I think we should eventually undo whatever change ends up in this PR. I'll remove the `ifdef` and lower the timeout.

Reply:
Changed in 822e8cf

Reply:
A yield may be more specific but our implementation of it is currently flawed, see WT-11940.